deploy_snapshot_and_javadoc.sh

```diff
@@ -24,4 +25,29 @@ else
   echo "Deploying snapshot..."
   ./gradlew clean uploadArchives
   echo "Snapshot deployed!"
+
+  echo "Uploading javadoc..."
+
+  ./gradlew aggregatedJavadocs
+
+  cd build/docs/javadoc
+  git init
+  git add .
+  git commit -m "Latest javadoc on successful travis build $TRAVIS_BUILD_NUMBER auto-pushed to gh-pages"
+
+  {
+    set +x
+    git config --global user.email "travis@travis-ci.org"
+    git config --global user.name "travis-ci"
+    git remote add origin "https://${GH_TOKEN}@github.com/wala/javadoc"
+    set -x
+  }
+
+  # we can force push here since we don't care about maintaining javadoc history for every commit
+  git push origin master --force
+
+  echo -e "Published Javadoc to gh-pages.\n"
+
+  cd $TRAVIS_BUILD_DIR
```

Comment:
Not sure what purpose this serves given that no further commands will be run.

Are you thinking that this script might be used in a `source` or `.` command rather than being run in its own shell? If so, then a more robust approach would be to enclose everything from the `cd build/docs/javadoc` line down to the `echo` in parentheses so that it runs in a subshell.